Doxyfile.in

```diff
@@ -0,0 +1,1552 @@
+# Doxyfile 1.6.2
```

Comment:
It is a new copy :(

I really didn't want to do that but it is essential for out of source builds to work (if you take a look in ``Doxyfile.in`` you'll see there are a bunch of ``@`` substitutions in there).

I'd rather there wasn't two. I'd argue that eventually we'd want to get rid of the ``Makefile`` build system all together, when we can drop having two Doxyfiles.